packages/providers/src/fetch_json.ts

```diff
@@ -60,10 +60,12 @@ export async function fetchJsonRpc(url: string, json: JsonRpcRequest, headers: o
             throw new ProviderError(await res.text(), { cause: status });
```

Comment:
So when status is other than `200`, the ok is false, so after we get `500`, `408` or another error, we stop here. So I suggest to move this line `if (!ok)` down the function

Also, have you tested your solution? If not, please test.